src/autosvg.js

```diff
@@ -0,0 +1,62 @@
+/* eslint-disable new-cap */
+import type { Layout } from "./engine";
+import Subview from "./subview";
+
+const transformers = {
+  rect: (layout: Layout): Object => {
+    return {
+      transformType: "props",
+      props: {
+        x: layout.left,
+        y: layout.top,
+        width: layout.width,
+        height: layout.height
+      }
+    };
+  },
+  line: (layout: Layout): Object => {
+    return {
+      transformType: "props",
+      props: {
+        x1: layout.left,
+        x2: layout.right,
+        y1: layout.top,
+        y2: layout.bottom
+      }
+    };
+  },
+  text: (layout: Layout): Object => {
+    return {
+      transformType: "props",
+      props: {
+        x: layout.left,
+        y: layout.top
+      }
+    };
+  },
+  circle: (layout: Layout): Object => {
+    return {
+      transformType: "props",
+      props: {
+        cx: layout.left + layout.width / 2,
+        cy: layout.top + layout.height / 2,
+        r: layout.width / 2
+      }
+    };
+  },
+  ellipse: (layout: Layout): Object => {
+    return {
+      transformType: "props",
+      props: {
+        cx: layout.left + layout.width / 2,
+        cy: layout.top + layout.height / 2,
+        rx: layout.width / 2,
+        ry: layout.height / 2
+      }
+    };
+  }
+};
+
+export default Object.keys(transformers).reduce((acc, key) => {
+  return {...acc, [key]: Subview(key, transformers[key])};
```

Comment:
GLOBAL: Mutate instead of extending where possible.

This:

```js
export default Object.keys(transformers).reduce((acc, key) => {
  return {...acc, [key]: Subview(key, transformers[key])};
}, {});
```

translates to:

```js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

exports.default = Object.keys(transformers).reduce(function (acc, key) {
  return _extends({}, acc, _defineProperty({}, key, Subview(key, transformers[key])));
}, {});
```

which is a *lot* of object extending / re-recreation.

Instead, doing a shallow straight mutation like:

```js
export default Object.keys(transformers).reduce((acc, key) => {
  acc[key] = Subview(key, transformers[key]);
  return acc;
}, {});
```

translates to:

```js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Object.keys(transformers).reduce(function (acc, key) {
  acc[key] = Subview(key, transformers[key]);
  return acc;
}, {});
```

and notably **avoids** all the object re-creation / extending...